File: Python/ExampleTests/production/IDForm.py
class IDForm:
    """Exactly five characters, beginning with Q."""

    def __init__(self, value):
        self.value = value if value is not None else ''
        if not (len(self.value) == 5 and self.value.startswith('Q')):
            raise ValueError('Must be 5 characters starting with Q')

    def __eq__(self, other):
        if not isinstance(other, IDForm):
            return NotImplemented
        return self.value == other.value

    def __hash__(self):
        return hash(self.value)

    def __str__(self):
        return f'IDForm{{{self.value}}}'

File: Java/SpectableJavaExamplesTest/src/main/java/production/IDForm.java
package production;

import java.util.Objects;

public class IDForm {
    public final String value;

    public IDForm(String value) {
        this.value = value != null ? value : "";
        if (!isValid())
            throw new NumberFormatException(
                    "Must be 5 characters starting with Q");
    }

    public boolean isValid() {
        return value.length() == 5 && value.startsWith("Q");
    }
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof IDForm)) return false;
        return Objects.equals(value, ((IDForm) o).value);
    }

    @Override
    public int hashCode() { return Objects.hash(value); }

    @Override
    public String toString() { return "IDForm{" + value + "}"; }
}
